Opera iOS post (#515)
ipfs/blog: content/post/opera-ios.md

> <p style="text-align: center;">

Reviewer: TODO: Front matter plz

Author: added!

  <img src="/img/opera-ipfs-header.png" alt="IPFS in Opera Touch for iOS">
</p>

> In 2020 we announced a big moment for IPFS: The first official support of IPFS protocol addressing in a major browser, when [Opera released IPFS support in their Android browser](ipns://blog.ipfs.io/2020-03-30-ipfs-in-opera-for-android/ "Opera Android IPFS announcement"). This was an important step in IPFS browser support generally, by building interest and momentum. We didn’t stop there, and we’ve got not just one… BUT TWO releases to share with you today:

Reviewer 2: I'd love to use `ipns://` by default, but we are not there yet :-D 
```suggestion
In 2020 we announced a big moment for IPFS: The first official support of IPFS protocol addressing in a major browser, when [Opera released IPFS support in their Android browser](https://blog.ipfs.io/2020-03-30-ipfs-in-opera-for-android/). This was an important step in IPFS browser support generally, by building interest and momentum. We didn’t stop there, and we’ve got not just one… BUT TWO releases to share with you today:
```

Author: OMG this is what happens when u have IPFS support by default in a browser ;)

First, Opera has now added support for IPFS addressing to Opera Touch, their iOS browser.

<p style="text-align: center;">
  <img src="/img/opera-ios-wikipedia-short.png" alt="A screenshot of Wikipedia on IPFS in Opera Touch">
</p>

Second, support for IPFS addressing in Opera desktop browser for Windows, macOS and Linux will be coming in their next release, currently planned for March 2021.

> With these releases, Opera will now support ipfs:// and ipns:// addressing across their browser product line on all major operating systems: Windows, macOS, Linux, iOS and Android.

Reviewer 2: ```suggestion
With these releases, Opera will now support `ipfs://` and `ipns://` addressing across their browser product line on all major operating systems: Windows, macOS, Linux, iOS and Android.
```

## How Do I use it?

First, install Opera Touch on your iOS device. If you’re reading this on an iOS device, [click to install now](https://apps.apple.com/us/app/opera-touch-web-browser/id1411869974)

<p style="text-align: center;">
  <img src="/img/opera-ios-app-store-short.png" alt="Opera Touch in the iOS app store">
</p>

> Then you can open content using IPFS protocol addresses, like the [ipns://blog.ipfs.io](blog you're reading now), [Wikipedia](ipns://en.wikipedia-on-ipfs.org/wiki/), or this [Persian room guardian](ipfs://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi/). You can easily upload files to IPFS using the [IPFS desktop application](https://docs.ipfs.io/install/ipfs-desktop/) or [Share.ipfs.io](https://share.ipfs.io].

Reviewer 2: I am afraid Hugo will ignore non-http URIs in markdown, but `<a>` tags should work?
Also, maybe shoutout to Anya?

```suggestion
Then you can open content using IPFS protocol addresses, like the <a href="ipns://blog.ipfs.io">blog you're reading now</a>, <a href="ipns://en.wikipedia-on-ipfs.org">Wikipedia</a>, or this <a href="ipfs://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi">Persian room guardian</a> by [Anya Boz](https://www.anyabozartist.com/the-persian-cat). You can easily upload files to IPFS using the [IPFS desktop application](https://docs.ipfs.io/install/ipfs-desktop/) or [Share.ipfs.io](https://share.ipfs.io).
```

## How Does it Work?

> Opera Touch supports navigating to addresses for ipfs:// and ipns:// protocol schemes, which are handled by a remote HTTP gateway. By default the gateway used is dweb.link, which is operated by Protocol Labs. TODO: Can select/change gateway?

Reviewer 2: ```suggestion
Opera Touch supports navigating to addresses for `ipfs://` and `ipns://` protocol schemes, which are handled by a remote HTTP gateway. By default the gateway used is `dweb.link`, which is operated by Protocol Labs. TODO: Can select/change gateway?
```

> Opera Touch supports navigating to addresses for `ipfs://` and `ipns://` protocol schemes, which are handled by a remote HTTP gateway. By default the gateway used is `dweb.link`, which is operated by Protocol Labs. TODO: Can select/change gateway?

Reviewer: TODO: Resolve this TODO line ☺️

Reviewer 3: Not in 2.5.5, should be released with 2.5.6 hopefully this week (release testing ongoing).

Author: Fixed!

> Native representation of IPFS addresses in browsers even when the content is loaded from an HTTP gateway as 1) it communicates to users the integrity guarantees built into the IPFS protocol and 2) it lets developers use IPFS addresses even as local node support is not yet implemented in most browsers.

Reviewer 2: ```suggestion
Native representation of IPFS addresses in browsers even when the content is loaded from an HTTP gateway as 1) it communicates to users the integrity guarantees built into the IPFS protocol and 2) it lets developers use IPFS addresses even as local node support is not yet implemented in most browsers.
```

I am not sure if I agree with this entire paragraph:
- (1) is plainly untrue (there are no integrity guarantees atm, user delegates trust to third party gateway). 
- (2) is is vague on added value. As Brave demonstrated you can resolve native URIs without keeping them in address bar. 

Can we rewrite it and  focus on less vague statements, like the fact that IPFS URIs are resolved out of the box, without the need for any additional extensions or opt-in settings, which familiarizes users and developers with decentralized concepts of content-addressing?

Author: Hm, we should figure out language for what integrity guarantees mean. I think we're talking about different things here. I'll remove entirely and replace with your suggestion for now!

## What’s next?

> Opera desktop browser support for IPFS addressing will ship soon, and we’re discussing what additional features to add next to build on top of this foundation of universal addressing support across the Opera browser line. Have ideas for what you’d like to see in Opera next? Let us know on the [IPFS forums](https://discuss.ipfs.io/), or tweet your ideas to [@Opera](https://twitter.com/opera) and [@IPFS](https://twitter.com/ipfs) on Twitter!

Reviewer 3: Anything about the preview builds on `ethdenver2021.opera.com`?

Author: Sure! Doesn't load yet?

Author: Added!
